py/ig_report_supernodes.py

```diff
@@ -63,27 +63,28 @@ def parse_abundance(s):
     parser.add_argument("--limit", "-l",
                         type=int,
                         default=5,
-                        help="abundance limit (default: %(default)s)")
+                        help="size limit (default: %(default)s)")
 
     args = parser.parse_args()
 
     print "Supernode reporter started..."
     print "Command line: %s" % " ".join(sys.argv)
 
-    result = []
-    with smart_open(args.input, "r") as fin:
+    input_size = output_size = 0
+    with smart_open(args.input, "r") as fin, smart_open(args.output, "w") as fout:
         for record in SeqIO.parse(fin, "fasta"):
-            abundance = parse_abundance(str(record.id))
-            if abundance >= args.limit:
-                old_id = str(record.id)
-                if "_abundance" in old_id:
-                    new_id = old_id.replace("_abundance:", "___size___")
-                else:
-                    new_id = "%s___size___%d" % (old_id, abundance)
-                record.id = record.name = record.description = new_id
-                result.append(record)
-
-    with smart_open(args.output, "w") as fout:
-        SeqIO.write(result, fout, "fasta")
-
+            input_size += 1
+            id = str(record.description)
+            size = parse_size(id)
+            if size >= args.limit:
+                if not "___size___" in id:
```

Comment:
Could this ever be false? And if yes shouldn't we update size? Maybe assert this if not supporting?